compiler/src/dmd/traits.d

```diff
@@ -1282,6 +1283,10 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
             //printf("getAttributes %s, attrs = %p, scope = %p\n", s.toChars(), s.userAttribDecl, s._scope);
             udad = s.userAttribDecl;
         }
+        else if (typeOfArg && !s)
```

Comment:
Is the `!s` required given it's an `else if` after `if (s)` ?

Reply:
No, but I wanted to be extra clear

Reply:
It doesn't make the code clearer. Someone looking at this code 1 year from now will think it's a leftover and want to remove it.

Reply:
Yeah, someone will want to hoist that isType into the if condition, avoid wasting compilation time if we can help it.